client/web/src/components/Auth/SignIn.js

```diff
@@ -9,20 +9,27 @@ import React, { useState } from 'react';
 import auth from '../../utils/firebase';
 import { signInWithPopup, GoogleAuthProvider } from "firebase/auth";
 import './styles.css';
+import {isIP, isIPv4} from 'is-ip';
 
 export const sendTokenToServer = async (token) => {
   // Send token to server
   const scheme = window.location.protocol;
   var currentHost = window.location.host;
   var parts = currentHost.split(':');
-  var ipAddress = parts[0];
+  var hostname = parts[0];
   var newPort = '8000';
-  var newHost = ipAddress + ':' + newPort;
+
+  if (hostname === 'localhost' || (isIP(hostname) || isIPv4(hostname))) {
```

Comment:
nit: It's best to only avoid having empty condition. e.g. here we can do `if (hostname !== 'localhost' && (isIP(hostname) && isIPv4(hostname))) {...}